client/my-sites/plugins/marketplace/marketplace-domain-upsell/index.tsx

```diff
@@ -0,0 +1,204 @@
+import ExternalLink from 'calypso/components/external-link';
+import MarketplaceShoppingCart from 'calypso/my-sites/plugins/marketplace/components/marketplace-shopping-cart';
+import theme from 'calypso/my-sites/plugins/marketplace';
+
+/**
+ * Style dependencies
+ */
+import 'calypso/my-sites/plugins/marketplace/marketplace-domain-upsell/style.scss';
+
+const MarketplaceHeaderTitle = styled.h1`
+	font-size: 2rem;
+	margin-bottom: 8px;
+`;
+const MarketplaceHeaderSubTitle = styled.h2`
+	font-size: 0.875rem;
+	margin-bottom: 15px;
+`;
+
+function MarketplaceDomainUpsellHeader() {
+	return (
+		<div>
+			<MarketplaceHeaderTitle className="marketplace-domain-upsell__title wp-brand-font">
+				{ translate( 'Choose a domain' ) }
+			</MarketplaceHeaderTitle>
+			<MarketplaceHeaderSubTitle>
+				{ translate(
+					'Yoast SEO requires a top level domain to index your site and help you rank higher.'
+				) }
+			</MarketplaceHeaderSubTitle>
+			<ExternalLink
+				href="https://yoast.com/domain-names-seo"
+				target="_blank"
+				rel="noopener noreferrer"
+				icon
+			>
+				{ translate( 'Domain names and their influence on SEO' ) }
+			</ExternalLink>
+			<HorizontalRule />
+		</div>
+	);
+}
+
+const getSiteNameFromURL = ( url: string ) => url?.split( '.' )?.[ 0 ] ?? url;
+
+function CalypsoWrappedMarketplaceDomainUpsell(): JSX.Element {
+	const [ selectedDomainProductUUID, setDomainProductUUID ] = useState< string >( '' );
+	const [ selectedDomain, setDomain ] = useState< DomainSuggestions.DomainSuggestion | null >(
+		null
+	);
+
+	const [ isExpandedBasketView, setIsExpandedBasketView ] = useState( false );
+	const { addProductsToCart, replaceProductsInCart, removeProductFromCart } = useShoppingCart();
+	const products = useSelector( getProductsList );
+	const previousPath = useSelector( getPreviousPath );
+	const isFetchingProducts = useSelector( isProductsListFetching );
+	const selectedSite = useSelector( getSelectedSite );
+	const { wpcom_url } = selectedSite;
+	const siteName = getSiteNameFromURL( wpcom_url );
+
+	useEffect( () => {
+		setIsExpandedBasketView( isDesktop() );
+	}, [ setIsExpandedBasketView ] );
+
+	useEffect( () => {
+		//FIXME: This code segment simulates yoast premium already being added when arriving here. To be removed when plugins page is completed.
+		if ( ! isFetchingProducts && products[ 'yoast_premium' ] ) {
+			const yoastProduct = fillInSingleCartItemAttributes(
+				{ product_slug: 'yoast_premium' },
+				products
+			);
+			replaceProductsInCart( [ yoastProduct ] );
+		}
+	}, [ isFetchingProducts, products, replaceProductsInCart ] );
+
+	const redirectToUseDomainFlow = (): void => {
+		const currentUrl = '/plugins/domain';
+		const useYourDomainUrl = `/start/new-launch/domains-launch/use-your-domain?siteSlug=${ selectedSite?.slug }&source=${ currentUrl }`;
+		page( useYourDomainUrl );
+	};
+
+	const onDomainSelect = ( suggestion: DomainSuggestions.DomainSuggestion ) => {
```

Comment:
Could you help me confirm if this can be replaced or simplified by the existing hook in https://github.com/Automattic/wp-calypso/blob/trunk/packages/launch/src/hooks/use-domain-selection.ts#L93? It's fine if it's not, but I'd like to confirm :)

Reply:
I hadn't noticed this, but seems like we need to show a live preview of adding to the basket. I don't think it can. be done with the specified approach.

Reply:
Cool. Let's stay with your implementation. Thanks for exploring this for me :)

Reply:
Cool. Let's stay with your implementation. Thanks for exploring this for me :)